sale_blanket_order/models/blanket_orders.py

```diff
@@ -264,9 +264,25 @@ def onchange_partner_id(self):
             values["team_id"] = self.partner_id.team_id.id
         self.update(values)
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        translated_draft = _("Draft")
+        for vals in vals_list:
+            company = vals.get("company_id") or self.env.company
```

Comment:
**issue**: Have you tried this? I do not believe that this will work, since `vals.get("company_id")` returns an integer, whereas `self.env.company` is a record. The following should work, but please do check it.

```suggestion
            company_id = vals.get("company_id")
            company = self.env["res.company"].browse(company_id) if company_id else self.env.company
```

Reply:
Confirmed on runboat, on saving a BO.

![image](https://github.com/OCA/sale-workflow/assets/78726989/4a9f5d89-b198-4a64-857f-7defade37890)